### ISSUE TITLE
Crashing when clicked on Customize Icons

### ISSUE THREAD

**Issue by Surendrajat** [otakuhqz/Omega#36]
## Description

Just crash...

## Expected Behavior

Should open icon customization view.

## Actual Behavior

App crashes.

## Steps to Reproduce

1. Long press on Home Screen.
2. Click on Home Settings.
3. Click on Theme Settings.
4. Click on Customize Icons.

## Device and OS

* Device: Oppo K3
* Android version: 10
* Launcher version: latest build from CI
* Rom: ColorOS v7.1

## Logcat

```
Build version: 0.6.3 
Build date: 1981-01-01 01:01:02 
Current date: 2021-01-09 19:14:29 
Device: OPPO CPH1955 
 
Stack trace:  
android.view.InflateException: Binary XML file line #22 in com.saggitt.omega:layout/fragment_icon_customization: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter iconShape
Caused by: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter iconShape
	at com.saggitt.omega.adaptive.IconShapeDrawable.<init>(Unknown Source:2)
	at com.saggitt.omega.icons.PreviewFrameView.loadBackground(PreviewFrameView.java:92)
	at com.saggitt.omega.icons.PreviewFrameView.onFinishInflate(PreviewFrameView.java:75)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1138)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1086)
	at android.view.LayoutInflater.parseInclude(LayoutInflater.java:1265)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1121)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1086)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:682)
	at com.saggitt.omega.OmegaLayoutInflater.inflate(OmegaLayoutInflater.kt:94)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:534)
	at com.saggitt.omega.icons.IconCustomizeFragment.onCreateView(IconCustomizeFragment.java:102)
	at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2698)
	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:320)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1187)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
	at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)
	at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:447)
	at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2169)
	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1992)
	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1947)
	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2629)
	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2577)
	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:247)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:541)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:210)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1433)
	at android.app.Activity.performStart(Activity.java:7986)
	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3528)
	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:226)
	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:206)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:178)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:102)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2229)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:238)
	at android.app.ActivityThread.main(ActivityThread.java:7798)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:995)

```

**Comment by 2br-2b** [comment]
I am still having this problem on the most recent release. Is there any chance of a new release coming out soon which will include the fix?

Thank you so much for your work! I love all the work you're putting into this and all of the customizati
on options possible!

**Comment by 2br-2b** [comment]
Device: Pixel 4XL
Android version: 11
Launcher version: latest version from Fdroid
Rom: GrapheneOS

**Comment by otakuhqz** [comment]
fixed in commit [81c8f52](https://github.com/otakuhqz/Omega/commit/81c8f52239db70f5a5383d4f6a17d08d5733d175)

**Comment by FridgeRacer** [comment]
I'm getting this crash on 0.8.1 (build 280, 17th of August 2021)

I installed the app from F-Droid using IzzyOnDroid's repository.
After, I also tried Omega Launcher Neo from your repository, and it gives the same crash (0.8.1-neo2)

Using Omega Launcher Alpha 0.9.0 build 317, the launcher just closes without an error message if I tap "Customize Icons"

`Build version: 0.8.1 
Build date: 1981-01-01 01:01:02 
Current date: 2022-06-26 06:25:01 
Device: OnePlus DN2103 
OS version: Android 11 (SDK 30) 
 
Stack trace:  
android.view.InflateException: Binary XML file line #24 in com.saggitt.omega:layout/fragment_icon_customization: Binary XML file line #19 in com.saggitt.omega:layout/desktop_preview: Error inflating class com.saggitt.omega.icons.PreviewFrameView
Caused by: android.view.InflateException: Binary XML file line #19 in com.saggitt.omega:layout/desktop_preview: Error inflating class com.saggitt.omega.icons.PreviewFrameView
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
	at android.view.LayoutInflater.createView(LayoutInflater.java:852)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1008)
	at android.view.LayoutInflater.parseInclude(LayoutInflater.java:1248)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1134)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1099)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:680)
	at com.saggitt.omega.OmegaLayoutInflater.inflate(OmegaLayoutInflater.kt:94)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:532)
	at com.saggitt.omega.icons.IconCustomizeFragment.onCreateView(IconCustomizeFragment.kt:63)
	at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2963)
	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:518)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3138)
	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:3072)
	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:251)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:502)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:246)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1442)
	at android.app.Activity.performStart(Activity.java:8175)
	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3851)
	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2414)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:265)
	at android.app.ActivityThread.main(ActivityThread.java:8360)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:632)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
Caused by: java.lang.SecurityException: Permission android.permission.READ_EXTERNAL_STORAGE denied for package com.saggitt.omega
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2389)
	at android.os.Parcel.createException(Parcel.java:2373)
	at android.os.Parcel.readException(Parcel.java:2356)
	at android.os.Parcel.readException(Parcel.java:2298)
	at android.app.IWallpaperManager$Stub$Proxy.getWallpaperWithFeature(IWallpaperManager.java:883)
	at android.app.WallpaperManager$Globals.getCurrentWallpaperLocked(WallpaperManager.java:494)
	at android.app.WallpaperManager$Globals.peekWallpaperBitmap(WallpaperManager.java:446)
	at android.app.WallpaperManager$Globals.peekWallpaperBitmap(WallpaperManager.java:422)
	at android.app.WallpaperManager.getDrawable(WallpaperManager.java:633)
	at com.saggitt.omega.wallpaper.WallpaperPreviewProvider.loadWallpaper(WallpaperPreviewProvider.kt:67)
	at com.saggitt.omega.wallpaper.WallpaperPreviewProvider.getWallpaper(WallpaperPreviewProvider.kt:55)
	at com.saggitt.omega.icons.PreviewFrameView.<init>(PreviewFrameView.kt:152)
	at com.saggitt.omega.icons.PreviewFrameView.<init>(PreviewFrameView.kt:37)
	at com.saggitt.omega.icons.PreviewFrameView.<init>(Unknown Source:6)
	... 36 more
Caused by: android.os.RemoteException: Remote stack trace:
	at android.os.storage.StorageManager.checkPermissionAndAppOp(StorageManager.java:1787)
	at android.os.storage.StorageManager.checkPermissionAndAppOp(StorageManager.java:1764)
	at android.os.storage.StorageManager.checkPermissionAndAppOp(StorageManager.java:1831)
	at android.os.storage.StorageManager.checkExternalStoragePermissionAndAppOp(StorageManager.java:1947)
	at android.os.storage.StorageManager.checkPermissionReadImages(StorageManager.java:1913)`
